spec/asciidoc-spec.coffee

```diff
@@ -16,11 +16,142 @@ describe "AsciiDoc grammar", ->
+      expect(tokens[6]).toEqual value: "; ,", scopes: ["source.asciidoc"]
+      expect(tokens[7]).toEqual value: "*bold*", scopes: ["source.asciidoc", "markup.bold.asciidoc"]
+
+    it "when variants of unbalanced asterisks around *bold* text", ->
+      {tokens} = grammar.tokenizeLine("*bold* **bold* ***bold* ***bold** ***bold*** **bold*** *bold*** *bold**")
+      expect(tokens.length).toEqual 14
+      expect(tokens[0]).toEqual value: "*bold*", scopes: ["source.asciidoc", "markup.bold.asciidoc"]
+      expect(tokens[1]).toEqual value: " ", scopes: ["source.asciidoc"]
+      expect(tokens[2]).toEqual value: "**bold* **", scopes: ["source.asciidoc", "markup.bold.asciidoc"]
+      expect(tokens[3]).toEqual value: "*bold* ", scopes: ["source.asciidoc"]
+      expect(tokens[4]).toEqual value: "***bold**", scopes: ["source.asciidoc", "markup.bold.asciidoc"]
+      expect(tokens[5]).toEqual value: " ", scopes: ["source.asciidoc"]
+      expect(tokens[6]).toEqual value: "***bold**", scopes: ["source.asciidoc", "markup.bold.asciidoc"]
+      expect(tokens[7]).toEqual value: "* ", scopes: ["source.asciidoc"]
+      expect(tokens[8]).toEqual value: "**bold**", scopes: ["source.asciidoc", "markup.bold.asciidoc"]
+      expect(tokens[9]).toEqual value: "* ", scopes: ["source.asciidoc"]
+      expect(tokens[10]).toEqual value: "*bold*", scopes: ["source.asciidoc", "markup.bold.asciidoc"]
+      expect(tokens[11]).toEqual value: "** ", scopes: ["source.asciidoc"]
+      expect(tokens[12]).toEqual value: "*bold*", scopes: ["source.asciidoc", "markup.bold.asciidoc"]
+      expect(tokens[13]).toEqual value: "*", scopes: ["source.asciidoc"]
+
+    it "when text is 'this is *bold* text'", ->
+      {tokens} = grammar.tokenizeLine("this is *bold* text")
+      expect(tokens.length).toEqual 3
+      expect(tokens[0]).toEqual value: "this is ", scopes: ["source.asciidoc"]
+      expect(tokens[1]).toEqual value: "*bold*", scopes: ["source.asciidoc", "markup.bold.asciidoc"]
+      expect(tokens[2]).toEqual value: " text", scopes: ["source.asciidoc"]
+
+    it "when text is '* text*'", ->
+      {tokens} = grammar.tokenizeLine("* text*")
+      expect(tokens.length).toEqual 3
+      expect(tokens[0]).toEqual value: "*", scopes: ["source.asciidoc", "markup.list.asciidoc", "markup.list.bullet.asciidoc"]
+      expect(tokens[1]).toEqual value: " ", scopes: ["source.asciidoc", "markup.list.asciidoc"]
+      expect(tokens[2]).toEqual value: "text*", scopes: ["source.asciidoc"]
+
+    it "when text is '*bold text*'", ->
+      {tokens} = grammar.tokenizeLine("*bold text*")
+      expect(tokens.length).toEqual 1
+      expect(tokens[0]).toEqual value: "*bold text*", scopes: ["source.asciidoc", "markup.bold.asciidoc"]
+
+    it "when text is '*bold*text*'", ->
+      {tokens} = grammar.tokenizeLine("*bold*text*")
+      expect(tokens.length).toEqual 1
+      expect(tokens[0]).toEqual value: "*bold*text*", scopes: ["source.asciidoc"]
```

Comment:
It seems this should render *bold*text* als bold, but somehow it doesn't. I'll have to look into this.